src/configure/test/suite/GitHubClient.test.ts

```diff
@@ -0,0 +1,135 @@
+import { GithubClient } from "../../clients/github/githubClient";
+
+var expect = require('chai').expect;
+var nock = require('nock');
+
+let org = "Organization";
+let repoName = "Repository";
+let githubClient = new GithubClient("PAT", "repoUrl");
+
+
+describe('# Testing listOrganizations() ', function () {
+    this.timeout(500000);
+
+    before(function () {
+        var orgList = [
+            {
+                "login": "SampleOrganizationA",
+                "id": 66720950,
+                "url": "https://api.github.com/orgs/SampleOrganizationA",
+                "repos_url": "https://api.github.com/orgs/SampleOrganizationA/repos"
+            },
+            {
+                "login": "SampleOrganizationB",
+                "id": 67440137,
+                "url": "https://api.github.com/orgs/SampleOrganizationB",
+                "repos_url": "https://api.github.com/orgs/SampleOrganizationB/repos",
+            },
+            {
+                "login": "SampleOrganizationC",
+                "id": 67453753,
+                "url": "https://api.github.com/orgs/SampleOrganizationC",
+                "repos_url": "https://api.github.com/orgs/SampleOrganizationC/repos"
+            }];
+
+        nock('https://api.github.com')
+            .get('/user/orgs')
+            .reply(200, orgList);
+    });
+
+    context('User is a member of organizations(s)', function () {
+        it('should return a list of organizations', function (done) {
+            githubClient.listOrganizations(true).then((orgs) => {
+                expect(Array.isArray(orgs)).to.equal(true);
+                expect(orgs.length).to.equal(3);
+                orgs.forEach((org) => {
+                    expect(org).to.have.property('login');
+                    expect(org).to.have.property('id');
+                    expect(org).to.have.property('url');
+                });
+                done();
+            });
+        });
+    });
+    before(function () {
+        var response = [];
+        nock('https://api.github.com')
+            .get('/user/orgs')
+            .reply(200, response);
+    });
+
+    context('User is not a member of any organization', function () {
```

Comment:
please add some negative cases - when user does not have permission

Reply:
Added.